src/app/hooks/useClickTracker/index.test.jsx

```diff
@@ -0,0 +1,174 @@
+/* eslint-disable react/prop-types */
+import React, { useContext } from 'react';
+import { renderHook } from '@testing-library/react-hooks';
+import { render, act } from '@testing-library/react';
+import useClickTracker from '.';
+import pidginData from './fixtureData/tori-51745682.json';
+import {
+  RequestContext,
+  RequestContextProvider,
+} from '#contexts/RequestContext';
+import {
+  ServiceContext,
+  ServiceContextProvider,
+} from '#contexts/ServiceContext';
+import { STORY_PAGE } from '#app/routes/utils/pageTypes';
+import * as params from '#containers/ATIAnalytics/params';
+import * as beacon from '#containers/ATIAnalytics/beacon';
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+const urlToObject = url => {
+  const { origin, pathname, searchParams } = new URL(url);
+
+  return {
+    origin,
+    pathname,
+    searchParams: Object.fromEntries(searchParams),
+  };
+};
+
+delete window.location;
+window.location = { href: 'http://bbc.com/pidgin/tori-51745682' };
+process.env.SIMORGH_ATI_BASE_URL = 'https://logws1363.ati-host.net?';
+
+const defaultProps = {
+  pageData: pidginData,
+  componentName: 'brand',
+};
+
+const WithContexts = ({ children }) => (
+  <RequestContextProvider
+    bbcOrigin="https://www.test.bbc.com"
+    pageType={STORY_PAGE}
+    isAmp={false}
+    service="pidgin"
+    pathname="/pidgin/tori-51745682"
+  >
+    <ServiceContextProvider service="pidgin">{children}</ServiceContextProvider>
+  </RequestContextProvider>
+);
+
+const TestComponentContainer = () => {
+  const clickRef = useClickTracker(defaultProps);
+  return (
+    <div>
+      <TestComponent ref={clickRef} />
+    </div>
+  );
+};
+
+const TestComponent = React.forwardRef((_, ref) => {
+  return (
+    <div id="test-component" ref={ref}>
+      <a href="https://www.test.bbc.com">Testing Link</a>
+      <button type="button" id="test-button">
+        Testing Button
+      </button>
+    </div>
+  );
+});
+
+// eslint-disable-next-line react/prop-types
+const wrapper = ({ children }) => <WithContexts>{children}</WithContexts>;
+
+describe('Click tracking', () => {
+  it('should return a ref', () => {
+    const { result } = renderHook(() => useClickTracker(defaultProps), {
+      wrapper,
+    });
+
+    const { container } = render(
+      <WithContexts>
+        <TestComponent ref={result.current} />
+      </WithContexts>,
+    );
+
+    expect(result.current.current).toBe(
+      container.querySelector('#test-component'),
+    );
+  });
+
+  it('should call buildATIClickParams with page data, requestContext, serviceContext', () => {
+    params.buildATIClickParams = jest.fn();
+
+    renderHook(() => useClickTracker(defaultProps), {
+      wrapper,
+    });
+
+    const requestContext = renderHook(() => useContext(RequestContext), {
+      wrapper,
+    });
+
+    const serviceContext = renderHook(() => useContext(ServiceContext), {
+      wrapper,
+    });
```

Comment:
Would you ever use the hook with just `RequestContext` or `ServiceContext`? Could just have one call here providing both contexts? Or am I misunderstanding the `renderHook` testing API?

Reply:
I haven't much experience with the `renderHook` API myself. I just tried this:
```js
const contexts = renderHook(() => {
    return {
      request: useContext(RequestContext),
      service: useContext(ServiceContext),
    };
});
```
It doesn't seem to get a value for either though, how were you thinking of combining the two?